app/src/main/java/com/android/shelfLife/viewmodel/recipes/IndividualRecipeViewModel.kt

```diff
@@ -0,0 +1,73 @@
+package com.android.shelfLife.viewmodel.recipes
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import com.android.shelfLife.model.recipe.Ingredient
+import com.android.shelfLife.model.recipe.Recipe
+import com.android.shelfLife.model.recipe.RecipeRepository
+
+class IndividualRecipeViewModel(
+    private val recipeRepository: RecipeRepository,
+) : ViewModel() {
+
+  var selectedRecipe by mutableStateOf<Recipe?>(null)
+  var selectedRecipeIsNonEmpty by mutableStateOf<Boolean>(true)
+
+  /**
+   * Initializes the ViewModel, setting the selected recipe from the repository. If no recipe is
+   * selected, it marks the recipe as empty.
+   */
+  init {
+    selectedRecipe = recipeRepository.selectedRecipe.value
+    if (selectedRecipe == null) {
+      selectedRecipeIsNonEmpty = false
+    }
+  }
+
+  fun getRecipeName(): String {
+    /**
+     * Returns the name of the selected recipe.
+     *
+     * @return The name of the recipe as a string.
+     */
+    return selectedRecipe!!.name
```

Comment:
## Error potential
Since selectedRecipe can be null by definition `Recipe?`, you should not force it with !!.
Instead, check if null
```suggestion
    return if selectedRecipe==null "" else selectedRecipe!!.name
```

Reply:
This applies for all the rest of the functions